pfhedge/nn/modules/hedger.py

```diff
@@ -284,27 +284,27 @@ def compute_hedge(
             tensor([[0.5056, 0.5295, 0.5845, 0.6610, 0.2918, 0.2918],
                     [0.5056, 0.3785, 0.4609, 0.5239, 0.7281, 0.7281]])
         """
-        self.inputs = self.inputs.of(derivative, self)
+        inputs = self.inputs.of(derivative, self)
         hedge = cast(List[Instrument], [derivative.ul()] if hedge is None else hedge)
 
         # Check that the spot prices of the hedges have the same sizes
         if not all(h.spot.size() == hedge[0].spot.size() for h in hedge):
             raise ValueError("The spot prices of the hedges must have the same size")
 
         (n_paths, n_steps), n_hedges = hedge[0].spot.size(), len(hedge)
-        if self.inputs.is_state_dependent():
+        if inputs.is_state_dependent():
             zeros = hedge[0].spot.new_zeros((n_paths, 1, n_hedges))
             save_prev_output(self, input=None, output=zeros)
             outputs = []
             for time_step in range(n_steps - 1):
-                input = self.get_input(time_step)  # (N, T=1, F)
-                outputs.append(self(input))  # (N, T=1, H)
+                input_ = inputs.get(time_step)  # (N, T=1, F)
+                outputs.append(self(input_))  # (N, T=1, H)
```

Comment:
nit: maybe can we just use input rather than input_ ?

Reply:
This is inside `for time_step in range(n_steps - 1):`. Thus, original `inputs` is used repeatedly and should not be overwritten.

Reply:
Thank you for the reply. I think a variable name `input` is not used elsewhere.

Reply:
Sorry, I was confused input <--> inputs. So, it was a misunderstanding. Now, I modified it.